src/connection_string.ts

```diff
@@ -204,6 +204,24 @@ function getUint(name: string, value: unknown): number {
   return parsedValue;
 }
 
+function toArray<T>(value: T): T[];
+function toArray<T>(value: T[]): T[];
+function toArray<T>(value: T | T[]): T[] {
+  return Array.isArray(value) ? value : [value];
+}
+
+function parseType(value: string): number | boolean | string {
+  try {
+    return getBoolean('', value);
+  } catch {
+    try {
+      return getInt('', value);
+    } catch {
+      return value;
+    }
+  }
+}
```

Comment:
I know I authored this but I'm thinking this is a 🦶 🔫  waiting to happen. `getBoolean` will give you true for `1` and false for `0` and `-1` making those numbers impossible to reach `getInt`. 
I think we should remove this, and instead inspect `authMechanismProperties` for the properties we know about, and translate them to their types and leave the rest as strings.